src/main/scala/shine/DPIA/FunctionalPrimitives/ForeignFunction.scala

```diff
@@ -64,7 +48,24 @@ final case class ForeignFunction(funDecl: ForeignFunction.Declaration,
         // with only one argument left to process return the assignment of the function call
         case Seq( (arg, inT) ) =>
           con(arg)(λ(expT(inT, read))(e =>
-            C( ForeignFunction(funDecl, inTs :+ inT, outT, exps :+ e) )) )
+            outT match {
+              // TODO: this is an ugly fix to avoid calling the function multiple times
+              //  for pair assignment
```

Comment:
Could we have an explicit link to the issue discussing this in the comment here?

Reply:
done